src/besselj.jl

```diff
@@ -150,3 +150,222 @@ function besselj1(x::Float32)
+
+        αp = evalpoly((1/x)^2, (s0, s1, s2, s3, s4, s5, s6, s7, s8, s9, s10, s11, s12, s13, s14, s15, s16, s17, s18, s19, s20, s21, s22, s23, s24, s25, s26, s27, s28))
+        α = x * (evalpoly((1/x)^2, (s0, -s1, -s2/3, -s3/5, -s4/7, -s5/9, -s6/11, -s7/13, -s8/15, -s9/17, -s10/19, -s11/21, -s12/23, -s13/25, -s14/27, -s15/29, -s16/31, -s17/33, -s18/35, -s19/37, -s20/39, -s21/41, -s22/43, -s23/45, -s24/47, -s25/49, -s26/51, -s27/53, -s28/55)))
+    else
+        αp = evalpoly((1/x)^2, (s0, s1, s2, s3, s4, s5, s6))
+        α = x * (evalpoly((1/x)^2, (s0, -s1, -s2/3, -s3/5, -s4/7, -s5/9, -s6/11)))
+    end
+    α = α - T(pi)/4 - T(pi)/2 * v
+    b = sqrt(2 / T(pi)) / sqrt(αp * x)
+
+    return α, b
+end
+# to acurrately use x > 1.7*nu
+function besselj_large_argument(v, x::T) where T
+    α, b = _α_αp_asymptotic(v, x)
+    return cos(α)*b
+end
+function bessely_large_argument(v, x::T) where T
+    α, b = _α_αp_asymptotic(v, x)
+    return sin(α)*b
+end
+
+# this needs a better way to sum these as it produces large errors
+# only valid in non-oscillatory regime (v>1/2, 0<t<sqrt(v^2 - 0.25))
+# power series has premature underflow for large orders
+# gives wrong answers for x > 20.0 (might want to fix this)
+function besselj_small_arguments_orders(v, x::T) where T
+    MaxIter = 1000
+    out = zero(T)
+    a = (x/2)^v / gamma(v + one(T))
```

Comment:
if you make this line `exp(muladd(log(x/2), v, -loggamma(v+1))`, this will work for both small and large `v`.

Reply:
I had originally chosen to break these up because the log version was sometimes less accurate.
```julia
## true value
julia> out = 2.515386282716736709635E-7
2.515386282716737e-7

julia> function besselj_small_arguments_orders(v, x::T) where T
           MaxIter = 1000
           out = zero(T)
           a = (x/2)^v / gamma(v + one(T))
           t2 = (x/2)^2
           for i in 0:MaxIter
               out += a
               abs(a) < eps(T) * abs(out) && break
               a *= -inv((v + i + one(T)) * (i + one(T))) * t2
           end
           return out
       end
besselj_small_arguments_orders (generic function with 1 method)

julia> 1 - out/besselj_small_arguments_orders(10.0, 2.0)
2.220446049250313e-16

julia> function besselj_small_arguments_orders(v, x::T) where T
           MaxIter = 1000
           out = zero(T)
           a = exp(muladd(log(x/2), v, -loggamma(v+1)))
           t2 = (x/2)^2
           for i in 0:MaxIter
               out += a
               abs(a) < eps(T) * abs(out) && break
               a *= -inv((v + i + one(T)) * (i + one(T))) * t2
           end
           return out
       end
besselj_small_arguments_orders (generic function with 1 method)

julia> 1 - out/besselj_small_arguments_orders(10.0, 2.0)
-1.9984014443252818e-15
```